applications/OptimizationApplication/python_scripts/controls/shape/vertex_morphing_shape_control.py

```diff
@@ -0,0 +1,363 @@
+            if not IsSameContainerExpression(physical_gradient, self.GetEmptyField()):
+                raise RuntimeError(f"Gradients for the required element container not found for control \"{self.GetName()}\". [ required model part name: {self.filter_model_part.FullName()}, given model part name: {physical_gradient.GetModelPart().FullName()} ]")
+
+            # now filter the field
+            filtered_gradient = self.filter.FilterIntegratedField(physical_gradient)
+
+            return filtered_gradient
+
+    def Update(self, new_control_field: ContainerExpressionTypes) -> bool:
+        if not IsSameContainerExpression(new_control_field, self.GetEmptyField()):
+            raise RuntimeError(f"Updates for the required element container not found for control \"{self.GetName()}\". [ required model part name: {self.filter_model_part.FullName()}, given model part name: {new_control_field.GetModelPart().FullName()} ]")
+        if KratosOA.ExpressionUtils.NormL2(self.control_field - new_control_field) > 1e-15:
+            with TimeLogger(self.__class__.__name__, f"Updating {self.GetName()}...", f"Finished updating of {self.GetName()}.",False):
+                # update the control SHAPE field
+                control_update = new_control_field - self.control_field
+                self.control_field = new_control_field
+                # now update the physical field
+                self._UpdateAndOutputFields(control_update)
+                # update explicit filter
+                if not self.is_filter_implicit:
+                    self.filter.Update()
+                return True
+        return False
+
+    def _UpdateMesh(self, shape_update) -> None:
+        if self.has_mesh_motion:
+            mm_computing_model_part = self._mesh_moving_analysis._GetSolver().GetComputingModelPart()
+            time_before_update = mm_computing_model_part.ProcessInfo.GetValue(Kratos.TIME)
+            step_before_update = mm_computing_model_part.ProcessInfo.GetValue(Kratos.STEP)
+            delta_time_before_update = mm_computing_model_part.ProcessInfo.GetValue(Kratos.DELTA_TIME)
+
+            # Reset step/time iterators such that they match the current iteration after calling RunSolutionLoop (which internally calls CloneTimeStep)
+            mm_computing_model_part.ProcessInfo.SetValue(Kratos.STEP, step_before_update-1)
+            mm_computing_model_part.ProcessInfo.SetValue(Kratos.TIME, time_before_update-1)
+            mm_computing_model_part.ProcessInfo.SetValue(Kratos.DELTA_TIME, 0)
+
+            # assign the boundary_displacements
+            Kratos.Expression.VariableExpressionIO.Write(shape_update, Kratos.MESH_DISPLACEMENT, True)
+
+            # solve for the volume mesh displacements
+            if not self._mesh_moving_analysis.time < self._mesh_moving_analysis.end_time:
+                self._mesh_moving_analysis.end_time += 1
+            self._mesh_moving_analysis.RunSolutionLoop()
```

Comment:
are you using V; to move solid nodes or MeshMotionApp and psuedo structure solution?

Reply:
we have "surface_implicit","surface_implicit_with_mesh_motion","bulk_surface_implicit","surface_explicit","surface_explicit_with_mesh_motion" filters, and those have mesh motion in name are using mesh motion which is psuedo structure solver